lib/src/equality.dart

```diff
@@ -212,7 +212,7 @@ abstract class _UnorderedEquality<E, T extends Iterable<E>>
   bool equals(T? elements1, T? elements2) {
     if (identical(elements1, elements2)) return true;
     if (elements1 == null || elements2 == null) return false;
-    var counts = HashMap(
+    var counts = HashMap<E, int?>(
```

Comment:
Yeah, but if you look down a few lines, we read from this as well:

```
var count = counts[e] ?? 0;
```

And `counts[e]` could return a null. I suppose I could use something like `Map.update()` to avoid the using a nullable type, but using a nullable type seems simpler here.

Reply:
The `null` is only when `e` isn't a present key in the map. We don't need the generic on the map to be nullable.

Reply:
Ah, yup, that's the case; I'll make this non-nullable.